### PR TITLE
Add ability to append ServerHttpRequest for kotlin-spring generator

### PR DESCRIPTION
**Pull request by Rugal**
Hi

I am creating `kotlin-spring` based spring-webflux application. I would like to access request header, but there is no way to achieve without adding function parameter. After checking all the available configuration and option, I found no feasible way to append an additional `ServerHttpRequest` to request handler, hence to add made this simple change in order to resolve the requirement.  

I Would like to discuss with people who interested in having request parameter in method.    
Since the one I append here is for `spring webflux` and `reactive`, therefore I put `ServerHttpRequest` which is a reactive only interface. Should I take `spring mvc` into consideration so that in that case the request object type should be `HttpServletRequest`?

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [@dr4ke616  @jimschubert 
](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR. please follow step 3 to update the samples and the doc.

**Comment by Rugal** [comment]
> thanks for the PR. please follow step 3 to update the samples and the doc.

step 3 is completed and doc is updated

**Comment by wing328** [comment]
Hi @Rugal thanks again for the PR. 

can you please add a config to enable/test this new option similar to ./bin/configs/kotlin-spring-boot-delegate.yaml ?

commit the new samples and then update .github/workflows/samples-kotlin-server.yaml with the new sample folder so that CI can test this feature moving forward?

**Comment by wing328** [comment]
I added the new sample files via https://github.com/OpenAPITools/openapi-generator/pull/17158/commits/45f8acc8dd21de692b2f030997b5c57641d8e5b8

Let's see if all the tests pass

**Comment by wing328** [comment]
the CI tests for the new sample failed: https://github.com/OpenAPITools/openapi-generator/actions/runs/7162598645/job/19499791008?pr=17158

can you please take a look when you've time?

**Comment by Rugal** [comment]
Will do once get some spare time, thanks for letting me know.

**Comment by wing328** [comment]
take your time. let me know if you need any help.

thanks again for the PR.

**Comment by Rugal** [comment]
Updated. Now the generated sample contains expected `ServerHttpRequest` parameter.

**Comment by Rugal** [comment]
For context and possible discussion detail @wing328 , please refer to the PR brief where I updated with some of my thoughts.

File: .github/workflows/samples-kotlin-server.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
           - samples/server/petstore/kotlin-springboot-delegate
           - samples/server/petstore/kotlin-springboot-modelMutable
           - samples/server/petstore/kotlin-springboot-reactive
+          - samples/server/petstore/kotlin-springboot-request
```
Comment (wing328) [inline .github/workflows/samples-kotlin-server.yaml:33]: looks like your force push removed some of the changes I made.

can yo remove this line and put it in .github/workflows/sampels-kotlin-servser-jdk17.yaml instead?

Comment (Rugal) [reply .github/workflows/samples-kotlin-server.yaml]: Sorry about that, didn't realize you pushed changed to my repository. Made a new commit for PR.

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/7181198352/job/20313881200?pr=17158

please update the samples and then i'll merge. thanks again for the PR.

**Comment by wing328** [comment]
merged. will update the samples with another commit.

**Comment by Rugal** [comment]
> Hi @Rugal the CI tests failed: https://github.com/OpenAPITools/openapi-generator/actions/runs/7526797968/job/20485651712
> 
> can you please take a look when you've time?

Sure, will do it ASAP

**Comment by Rugal** [comment]
Hi @wing328,

A fix is ready [here](https://github.com/OpenAPITools/openapi-generator/pull/17623)

**Comment by defaultbranch** [comment]
@Rugal test

**Comment by defaultbranch** [comment]
@Rugal Thanks for your efforts!

Just today I discovered that there is a `kotlin-spring` generator, and after using the [`spring`](https://openapi-generator.tech/docs/generators/spring/) generator for >1 year, I thought I'd give [`kotlin-spring`](https://openapi-generator.tech/docs/generators/kotlin-spring) a try.

(My motivation is, I want to use `<reactive>true</reactive>`, so far this meant [project reactor](https://projectreactor.io/) which is great but only works at the framework level (it has to cope with the limitations of the Java language); Kotlin supports the same thing at compiler-level (thanks to its co-routines), which simplifies the code a great deal; and I was already happy to see that the generator option `<reactive>true</reactive>` is actually the same for the `spring` (Java with Project Reactor) and `kotlin-spring` (Kotlin co-routines))

Today I tried the `kotlin-spring` generator (instead of `spring` generator) for the first time; and I figured out that the option `<appendRequestToHandler>true</appendRequestToHandler>` does nothing much (only adds an import statement).

Long story short, I have a comment:

- instead of `ServerHttpRequest`, I was so far using `ServerWebExchange` (provided by the `spring` generator)

and two questions:

1. why do you go for `ServerHttpRequest` instead of `ServerWebExchange`? Wouldn't the later be closer to what the `spring` generator already provides?
2. given that both `ServerHttpRequest` and `ServerWebExchange` serve a similar purpose, what is the status of your `<appendRequestToHandler>`, what are the remaining obstacles to get this functionality into the `kotlin-spring` generator?

**Comment by Rugal** [comment]
Hi @defaultbranch ,

My initial motivation for this change is that I would like to access request header information in handler/operation. Because I was using `webflux`, which is a reactive framework, while still want to be able to access request object.  
After some search, I found `ServerHttpRequest` is quite a good fit for my need.  
The `appendRequestToHandler` parameter should be able to generate an import, but also to append the `ServerHttpRequest` to method signature as well.  
Your suggestion is great, I think we can replace the request with exchange, which contains both request and response.
